mami/script.js

```diff
@@ -0,0 +1,123 @@
+// # 48-PB-datastructure-classes
+// # Classy
+
+// #### 1. Person Class
+// * Create a class called `Person` which accepts the name of a person as a string, and his/her age as a number.
+// The `Person` class should have a method called describe which returns a string with the following syntax: "name, age years old". So,
+// for example, if John is 19 years old, then the function describe of his object will return "John, 19 years old".
+class Person {
+  constructor(name, age) {
+    this.name = name;
+    this.age = age;
+  }
+
+  describe() {
+    return `${this.name},${this.age} years old`;
+  }
+}
+
+let prof = new Person("John", 19);
+console.log(prof);
+
+console.log(prof.describe());
+// #### 2. Volume
+// * Write a JavaScript program to get the volume of a cylinder with four decimal places using a class.
+class Volume {
+  constructor(r, h) {
+    this.r = r;
+    this.h = h;
+  }
+
+  calc() {
+    return (Math.PI * this.r ** 2 * this.h).toFixed(4);
+  }
+}
+
+let vol = new Volume(2, 4);
+console.log(vol.calc());
+// * **Note**: Volume of a cylinder : `V = π(Math.PI)r^2h` - r is the radius and h is the height of the cylinder.
+
+// #### 3. Tick Tock
+// Rewrite the following code to use the "class" syntax.
+// ```javascript
+
+class Clock {
+  constructor({ template }) {
+    this.template = template;
+  }
+  render() {
+    let date = new Date();
+
+    let hours = date.getHours();
+    if (hours < 10) hours = "0" + hours;
+
+    let mins = date.getMinutes();
+    if (mins < 10) mins = "0" + mins;
+
+    let secs = date.getSeconds();
+    if (secs < 10) secs = "0" + secs;
+
+    let output = template
+      .replace("h", hours)
+      .replace("m", mins)
+      .replace("s", secs);
+
+    console.log(output);
+  }
+  stop() {
+    clearInterval(this.timer);
+  }
+
+  start() {
+    this.render();
+    this.timer = setInterval(this.render, 1000);
```

Comment:
Task 3 ==>    You will face error here ==>   this.timer = setInterval(this.render, 1000);
this.timer must be ==> 
  this.timer = setInterval(() => {
        this.render();
      }, 1000);